File: js/ejer_37.js
(async function() {
	function hasPrompt() { return typeof prompt === 'function'; }

	function askBrowser(q) { return prompt(q); }

	function askNode(q) {
		const rl = require('readline').createInterface({ input: process.stdin, output: process.stdout });
		return new Promise(resolve => rl.question(q, ans => { rl.close(); resolve(ans); }));
	}

	async function ask(q) {
		if (hasPrompt()) return askBrowser(q);
		if (typeof require === 'function' && typeof process !== 'undefined') return await askNode(q);
		return null;
	}

	const rawN = await ask('Ingrese la cantidad de términos N para Fibonacci: ');
	const N = parseInt(rawN, 10);
	if (isNaN(N) || N <= 0) {
		console.log('Entrada inválida. Introduce un entero positivo para N.');
		return;
	}

	const fib = [];
	for (let i = 0; i < N; i++) {
		if (i === 0) fib.push(0);
		else if (i === 1) fib.push(1);
		else fib.push(fib[i-1] + fib[i-2]);
	}

	console.log('Fibonacci hasta', N, 'términos:', fib.join(', '));
	if (typeof document !== 'undefined') {
		const pre = document.createElement('pre');
		pre.textContent = 'Fibonacci hasta ' + N + ' términos: ' + fib.join(', ');
		document.body.appendChild(pre);
	}
})();

File: python/ejer_37.py
def pedir_entero(prompt_text):
	while True:
		try:
			v = int(input(prompt_text))
			return v
		except ValueError:
			print('Entrada inválida. Introduce un entero.')


def fibonacci_n_terms(n):
	if n <= 0:
		return []
	if n == 1:
		return [0]
	seq = [0, 1]
	while len(seq) < n:
		seq.append(seq[-1] + seq[-2])
	return seq


def main():
	N = pedir_entero('Ingrese la cantidad de términos N para Fibonacci: ')
	if N <= 0:
		print('N debe ser un entero positivo.')
		return
	seq = fibonacci_n_terms(N)
	print(f'Fibonacci hasta {N} términos: {", ".join(str(x) for x in seq)}')


if __name__ == '__main__':
	main()

